docs/internal/DistributedArchitectureGuide.md

```diff
@@ -144,15 +144,74 @@ Some concepts are applicable to both cluster and project scopes, e.g. [persisten
 
 ### Translog
 
-(Explain checkpointing and generations, when happens on Lucene flush / fsync)
+[Basic write model]:https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-replication.html
 
-(Concurrency control for flushing)
+It is important to understand first the [Basic write model] of documents:
+documents are written to Lucene in-memory buffers, then "refreshed" to searchable segments which may not be persisted on disk, and finally "flushed" to a durable Lucene commit on disk.
+This means that newly ingested data may be lost if there is an outage before the next persisted commit on disk.
+For this reason, newly ingested data is also written to a shard's `[Translog](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/translog/Translog.java)`, whose main purpose is to persist uncommitted operations (e.g., document insertions or deletions), so they can be replayed in the event of ephemeral failures such as a crash or power loss.
+The translog is always persisted and fsync'ed on disk before acknowledging writes back to the user.
+This can be seen in `[InternalEngine](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java)` which calls the `add()` method of the translog to append operations, e.g., its `index()` method at some point adds a document insertion operation to the translog.
+The translog ultimately truncates operations once they have been flushed to disk by a Lucene commit.
 
-(VersionMap)
+Main usages of the translog are:
+
+* During recovery, an index shard can be recovered up to at least the last acknowledged operation by replaying the translog onto the last flushed commit of the shard.
+* Facilitate real-time (m)GETs of documents without refreshing.
 
 #### Translog Truncation
 
-#### Direct Translog Read
+[Flush API]:https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-flush.html
+
+Translog files are automatically truncated when they are no longer needed, specifically after all their operations have been persisted by Lucene commits on disk.
+Lucene commits are initiated by flushes (e.g., with the index [Flush API]).
+
+Flushes may also be automatically initiated by Elasticsearch, e.g., if the translog exceeds a configurable size ([`INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING`](https://github.com/elastic/elasticsearch/blob/dd1db5031ee7fdac284753c0c3b096b0e981d71a/server/src/main/java/org/elasticsearch/index/IndexSettings.java#L352)) or age ([`INDEX_TRANSLOG_FLUSH_THRESHOLD_AGE_SETTING`](https://github.com/elastic/elasticsearch/blob/dd1db5031ee7fdac284753c0c3b096b0e981d71a/server/src/main/java/org/elasticsearch/index/IndexSettings.java#L370)), which ultimately truncates the translog as well.
+
+#### Acknowledging writes
```

Comment:
Is it worth mentioning here that we do this to amortize the cost of `fsync()` operations across all the writes?